test/configSpec.js

```diff
@@ -160,6 +160,31 @@ describe('Config', () => {
       expect(() => Config.loadConfig()).to.throw(ServerError)
     })
 
+    it('PUBLIC_URI', () => {
+      process.env.UNIT_TEST_OVERRIDE = 'true'
+      process.env.PUBLIC_URI = 'https://example.com:1234/ledger/path/'
+      const _config = Config.loadConfig()
+      expect(_config.getIn(['server', 'public_secure'])).to.equal(true)
+      expect(_config.getIn(['server', 'public_host'])).to.equal('example.com')
+      expect(_config.getIn(['server', 'public_port'])).to.equal(1234)
+      expect(_config.getIn(['server', 'public_path'])).to.equal('/ledger/path/')
+      expect(_config.getIn(['server', 'base_uri'])).to.equal(
+        'https://example.com:1234/ledger/path/')
+    })
+
+    it('PUBLIC_URI - no port or path', () => {
+      process.env.UNIT_TEST_OVERRIDE = 'true'
+      process.env.PUBLIC_URI = 'http://www.example.com'
+      const _config = Config.loadConfig()
+      expect(_config.getIn(['server', 'public_secure'])).to.equal(false)
+      expect(_config.getIn(['server', 'public_host']))
+        .to.equal('www.example.com')
+      expect(_config.getIn(['server', 'public_port'])).to.equal(80)
```

Comment:
Might be worth adding a test for a https URI to make sure it defaults to port 443 there. But definitely a nice-to-have. :)